test/e2e/dr/quorum-restore.go

```diff
@@ -0,0 +1,330 @@
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e2e.Logf("Update DNS records")
+		var survivingMasterIP string
+		runCommandAndRetry("easy_install --user pip && ~/.local/bin/pip install --user boto3")
+
+		infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		internalAPI, err := url.Parse(infra.Status.APIServerURL)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		internalAPI.Host = strings.Replace(internalAPI.Host, "api.", "", 1)
+
+		domain, _, err := net.SplitHostPort(internalAPI.Host)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		e2e.Logf("domain: %s", domain)
+		masterNodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/master",
+		})
+		for i := range masterNodes.Items {
+			node := &masterNodes.Items[i]
+			etcdName := fmt.Sprintf("etcd-%d.%s", i, domain)
+			masterIP := ""
+			for _, address := range node.Status.Addresses {
+				if address.Type == "InternalIP" {
+					masterIP = address.Address
+					break
+				}
+			}
+			if node.GetName() == survivingNodeName {
+				survivingMasterIP = masterIP
+			}
+			updateDNS(domain, etcdName, masterIP)
+		}
+
+		imagePullSecretPath := getPullSecret(oc)
+		defer os.Remove(imagePullSecretPath)
+		runPodSigner(oc, survivingNodeName, imagePullSecretPath, proxy)
+
+		e2e.Logf("Restore etcd on remaining masters")
+		setupEtcdEnvImage := getImagePullSpecFromRelease(oc, imagePullSecretPath, "setup-etcd-environment")
+		kubeClientAgent := getImagePullSpecFromRelease(oc, imagePullSecretPath, "kube-client-agent")
+		for i := range masterNodes.Items {
+			node := &masterNodes.Items[i]
+			masterDNS := ""
+			for _, address := range node.Status.Addresses {
+				if address.Type == "InternalDNS" {
+					masterDNS = address.Address
+					break
+				}
+			}
+			if masterDNS == survivingNodeName {
+				e2e.Logf("Skipping node as its the surviving master")
+				continue
+			}
+			runViaBastionSSH(masterDNS, proxy,
+				fmt.Sprintf("sudo -i env SETUP_ETCD_ENVIRONMENT=%s KUBE_CLIENT_AGENT=%s /bin/bash -x /usr/local/bin/etcd-member-recover.sh %s \"etcd-member-%s\"",
+					setupEtcdEnvImage, kubeClientAgent, survivingMasterIP, node.GetName()))
+		}
+
+		e2e.Logf("Wait for etcd pods to become available")
+		_, err = exutil.WaitForPods(
+			oc.AdminKubeClient().CoreV1().Pods("openshift-etcd"),
+			exutil.ParseLabelsOrDie("k8s-app=etcd"),
+			exutil.CheckPodIsReady,
+			expectedNumberOfMasters,
+			10*time.Minute,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		scaleEtcdQuorum(oc, expectedNumberOfMasters)
+
+		e2e.Logf("Remove etcd signer")
+		err = oc.AdminKubeClient().CoreV1().Pods("openshift-config").Delete("etcd-signer", &metav1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		restartSDNPods(oc)
+		waitForMastersToUpdate(oc, mcps)
+		waitForOperatorsToSettle(coc)
+	})
+})
+
+func scaleEtcdQuorum(oc *exutil.CLI, replicas int) {
```

Comment:
it seems like this is a thing a customer would always have to run.

Reply:
This is not necessary - etcdQuorum DS would be unable to schedule pods on nodes, but afaik it doesn't affect any operator